sirepo/template/opal.py

```diff
@@ -448,10 +489,79 @@ def sim_frame_plot2Animation(frame_args):
 def write_parameters(data, run_dir, is_parallel):
     pkio.write_text(
         run_dir.join(OPAL_INPUT_FILE),
-        _generate_parameters_file(data),
+        _generate_parameters_file(data, is_parallel),
     )
 
 
+class _Generate(template_common.LibLatticeUtilMixin):
```

Comment:
base class unless there's a reason to do mixin, which I don't see. Also, this doesn't belong in tempate_common. Put it in sirepo.lib since nobody else would use this.

Reply:
c4d0d0bf